packages/medusa/src/services/inventory.ts

```diff
@@ -0,0 +1,93 @@
+import { BaseService } from "medusa-interfaces"
+import { MedusaError } from "medusa-core-utils"
+import { TransactionBaseService } from "../interfaces"
+import { EntityManager } from "typeorm"
+import ProductVariantService from "./product-variant"
+import { ProductVariant } from "../models"
+
+type InventoryServiceProps = {
+  manager: EntityManager
+  productVariantService: ProductVariantService
+}
+class InventoryService extends TransactionBaseService<InventoryService> {
+  protected productVariantService_: ProductVariantService
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  constructor({ manager, productVariantService }: InventoryServiceProps) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.productVariantService_ = productVariantService
+  }
+
+  /**
+   * Updates the inventory of a variant based on a given adjustment.
+   * @param variantId - the id of the variant to update
+   * @param adjustment - the number to adjust the inventory quantity by
+   * @return resolves to the update result.
+   */
+  async adjustInventory(
+    variantId: string | undefined | null,
```

Comment:
question: Should we really be allowed to pass undefined or null here?

I think that it should only be string and update the consumer